### PR TITLE
TEST - CSIPSTR5 test case and packages.

### LINKED ISSUE #23: CSIPSTR5 Test Case Description
**Specification:**
- **Name:** E-ARK CSIP
  **Version:** 2.0-DRAFT
  **URL:** http://earkcsip.dilcis.eu/
**:**
**Specification:** CSIP

**Version:** 2.0-DRAFT

**Requirement:**
- **Id:** CSIPSTR5
- **Link:** http://earkcsip.dilcis.eu/#CSIPSTR5

**Corpus Folder:** `corpora/csip/structure/folders/`

**Error Level:** ERROR

**Description:**
> The Information Package folder MUST include a folder named metadata, which MUST include at least all metadata relevant for the whole package

The second part of this requirement needs rethinking:
> at least all metadata relevant for the whole package

is subjective, who judges what is all relevant metadata and I’m not sure that this specification wants to get dragged in to judging metadata completeness and quality.

### PR DESCRIPTION
**Pull request by carlwilson**
Factored out from original work by @ltaimre

closes #23

### CODE REVIEW

**Review by ltaimre — DISMISSED** [review]

File: corpus/structure/CSIPSTR5/IP_18000_CSIPSTR5_1/METS.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!-- Minimal IP with schemas -->
+<!-- In this example IDs are carring information to ease understanding - they look like paths to ease understanding, but are just IDs, similar to the naming of namespaces - these IDs can be replaced with information less UUID -->
+<!-- CSIPSTR15 goes like this: We recommend including all schema documents for any structured metadata within package. These schema documents SHOULD be placed in a sub-folder called schemas within the Information Package root folder and/or the representation folder. -->
+<mets xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+  xmlns="http://www.loc.gov/METS/" 
+  xmlns:csip="https://DILCIS.eu/XML/METS/CSIPExtensionMETS"
+  xmlns:xlink="http://www.w3.org/1999/xlink"
+  xsi:schemaLocation="http://www.w3.org/2001/XMLSchema-instance schemas/XMLSchema.xsd http://www.loc.gov/METS/ schemas/mets.xsd http://www.w3.org/1999/xlink schemas/xlink.xsd https://DILCIS.eu/XML/METS/CSIPExtensionMETS schemas/CSIPExtensionMETS.xsd" 
+  OBJID="Minimal_IP_with_schemas"
```
Comment (PhillipTommerholt) [inline corpus/structure/CSIPSTR5/IP_18000_CSIPSTR5_1/METS.xml]: wrong objid

File: corpus/structure/CSIPSTR5/IP_18000_CSIPSTR5_10/METS.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!-- Minimal IP with schemas -->
+<!-- In this example IDs are carring information to ease understanding - they look like paths to ease understanding, but are just IDs, similar to the naming of namespaces - these IDs can be replaced with information less UUID -->
+<!-- CSIPSTR15 goes like this: We recommend including all schema documents for any structured metadata within package. These schema documents SHOULD be placed in a sub-folder called schemas within the Information Package root folder and/or the representation folder. -->
+<mets xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+  xmlns="http://www.loc.gov/METS/" 
+  xmlns:csip="https://DILCIS.eu/XML/METS/CSIPExtensionMETS"
+  xmlns:xlink="http://www.w3.org/1999/xlink"
+  xsi:schemaLocation="http://www.w3.org/2001/XMLSchema-instance schemas/XMLSchema.xsd http://www.loc.gov/METS/ schemas/mets.xsd http://www.w3.org/1999/xlink schemas/xlink.xsd https://DILCIS.eu/XML/METS/CSIPExtensionMETS schemas/CSIPExtensionMETS.xsd" 
+  OBJID="Minimal_IP_with_schemas"
```
Comment (PhillipTommerholt) [inline corpus/structure/CSIPSTR5/IP_18000_CSIPSTR5_10/METS.xml]: wrong objid

**Review by PhillipTommerholt — CHANGES_REQUESTED** [review]
All OBJIDs in the packages should be adjusted to the proper name.

File: corpus/structure/CSIPSTR5/testCase.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    <!-- Individual validation rule entry, identified by a running number that's unique within the test case only -->
+    <!-- Rule for the existence of a dmdSec element in mets.xml -->
+    
+    <rule id="1">
+      <!-- Text description of the validation rule -->
+      <description>The Information Package root folder SHOULD include a folder named metadata.</description>
+      <!-- Details of the validation error generated by the message, can be one of: [ ERROR | WARNING | INFO ]-->
+      <error level="WARNING">
+        <!-- Expected error message when the rule is violated. -->
+        <message>The Information Package folder does not include folder named metadata. It MUST include a folder named metadata </message>
```
Comment (koit) [inline corpus/structure/CSIPSTR5/testCase.xml:29]: The correct is "It SHOULD include …" vs the current "It MUST include …".

File: corpus/structure/CSIPSTR5/testCase.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
+<!-- Root element for an individual test case, allows these to be wrapped into
+XML lists -->
+<testCase xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:noNamespaceSchemaLocation="testCase.xsd" testable="TRUE">
+  <!-- Unique ID for the test case -->
+  <id specification="E-ARK CSIP" version="2.0-DRAFT" requirementId="CSIPSTR5"/>
```
Comment (koit) [inline corpus/structure/CSIPSTR5/testCase.xml:6]: Version should be `version="2.0.4"`.

**Review by PhillipTommerholt — APPROVED** [review]

File: corpus/structure/CSIPSTR5/IP_18000_CSIPSTR5_1/METS.xml (hunk 1 above)
Comment (koit) [reply corpus/structure/CSIPSTR5/IP_18000_CSIPSTR5_1/METS.xml]: Related to `mets/@OBJID`, according to CSIP86, `mets/structMap[@LABEL='CSIP']/div/@LABEL` must also match the name of the package root folder.

**Review by koit — APPROVED** [review]
All good now with the packages. testCase.xml could use the two tiny tweaks that I pointed out.

**Pull request merged**